homeassistant/components/yale_smart_alarm/coordinator.py

```diff
@@ -0,0 +1,138 @@
+                    continue
+                if (
+                    lock_status
+                    and (
+                        "device_status.lock" in state or "device_status.unlock" in state
+                    )
+                    and closed
+                    and locked
+                ):
+                    device["_state"] = "locked"
+                    locks.append(device)
+                    continue
+                if (
+                    lock_status
+                    and (
+                        "device_status.lock" in state or "device_status.unlock" in state
+                    )
+                    and closed
+                    and not locked
+                ):
+                    device["_state"] = "unlocked"
+                    locks.append(device)
+                    continue
+                if (
+                    lock_status
+                    and (
+                        "device_status.lock" in state or "device_status.unlock" in state
+                    )
+                    and not closed
+                ):
+                    device["_state"] = "unlocked"
+                    locks.append(device)
+                    continue
+                device["_state"] = "unavailable"
+                locks.append(device)
+                continue
+            if device["type"] == "device_type.door_contact":
+                if "device_status.dc_close" in state:
+                    device["_state"] = "closed"
+                    door_windows.append(device)
+                    continue
+                if "device_status.dc_open" in state:
+                    device["_state"] = "open"
+                    door_windows.append(device)
+                    continue
+                device["_state"] = "unavailable"
+                door_windows.append(device)
+                continue
+
+        return {
+            "alarm": updates["arm_status"],
+            "locks": locks,
+            "door_windows": door_windows,
+            "status": updates["status"],
+            "online": updates["online"],
+        }
+
+    def get_updates(self) -> dict:
+        """Fetch data from Yale."""
+
+        if self.yale is None:
+            self.yale = YaleSmartAlarmClient(
+                self.entry.data[CONF_USERNAME], self.entry.data[CONF_PASSWORD]
+            )
+
+        try:
+            arm_status = self.yale.get_armed_status()
+            cycle = self.yale.get_cycle()
+            status = self.yale.get_status()
+            online = self.yale.get_online()
+
+        except AuthenticationError as error:
+            LOGGER.error("Authentication failed. Check credentials %s", error)
+            self.hass.async_create_task(
+                self.hass.config_entries.flow.async_init(
+                    DOMAIN,
+                    context={"source": SOURCE_REAUTH, "entry_id": self.entry.entry_id},
+                    data=self.entry.data,
+                )
+            )
```

Comment:
I think you need to return here, otherwise you risk getting partially through your earlier requests, catching an exception, and then trying to pass variables that haven't been set yet to the return

Reply:
Probably raise `UpdateFailed`.

Reply:
Makes sense. Will do it now